### PR TITLE
ci: depends: add script that download and unpack SDL/SDL_mixer libs f…

### PR DESCRIPTION
**Pull request by dalerank**
…rom command line

### CODE REVIEW

**Comment by dalerank** [comment]
Fixed linux compilation

**Comment by dalerank** [comment]
Fixed linux compilaton/disabled win32 stacktrace

**Comment by crudelios** [comment]
I like this very much, however platform dependent code should be in the `platform` directory, not in `core`. Can you move `backtrace.c` from `core` to `platform`, please?

**Comment by dalerank** [comment]
done

**Comment by dalerank** [comment]
@Keriew  Any changes ? all error fixed

File: .ci_scripts/install_dependencies.bat [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cd ..
+mkdir tmp
+cd tmp
+curl -o SDL2-devel-2.0.12-VC.zip https://www.libsdl.org/release/SDL2-devel-2.0.12-VC.zip
```
Comment (crudelios) [inline .ci_scripts/install_dependencies.bat:4]: Why 2.0.12? SDL is currently at version 2.0.16 already.

File: CMakeLists.txt [hunk 2]
```diff
@@ -229,9 +229,20 @@ elseif (${TARGET_PLATFORM} STREQUAL "android")
     )
 endif()
 
+if (WIN32)
+    set(PLATFORM_FILES
+        ${PLATFORM_FILES}
+        ${PROJECT_SOURCE_DIR}/src/platform/windows/backtrace_windows.c
```
Comment (crudelios) [inline CMakeLists.txt:235]: Nitpick: `windows/backtrace_windows.c` (and `posix/backtrace_posix.c`) seems redundant. I think `windows/backtrace.c` and `posix/backtrace.c` is enough.

File: src/platform/platform.c [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
     return SDL_VERSIONNUM(v.major, v.minor, v.patch) >= SDL_VERSIONNUM(major, minor, patch);
 }
+
+void exit_with_status(int status)
+{
+#ifdef __EMSCRIPTEN__
+    EM_ASM(Module.quitGame($0), status);
+#endif
+    exit(status);
+}
```
Comment (crudelios) [inline src/platform/platform.c:19]: Newline at end of file.

File: .ci_scripts/install_dependencies.bat [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+curl -o SDL2_mixer-devel-2.0.4-VC.zip https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-devel-2.0.4-VC.zip
+tar -xf SDL2-devel-2.0.12-VC.zip -C ../ext/SDL2
+tar -xf SDL2_mixer-devel-2.0.4-VC.zip -C ../ext/SDL2
+cd ../.ci_scripts
```
Comment (crudelios) [inline .ci_scripts/install_dependencies.bat:8]: Newline at end of file.

File: src/platform/posix/backtrace_posix.c [hunk 5]
```diff
@@ -1,12 +1,15 @@
-#include "backtrace.h"
+#include "platform/backtrace.h"
+#include "platform/platform.h"
 
+#include "SDL.h"
```
Comment (crudelios) [inline src/platform/posix/backtrace_posix.c:4]: External includes should be after internal includes (`SDL.h` should come after).

File: src/platform/windows/backtrace_windows.c [hunk 6]
```diff
@@ -0,0 +1,143 @@
+#include "platform/backtrace.h"
+
+#include "SDL.h"
```
Comment (crudelios) [inline src/platform/windows/backtrace_windows.c:3]: External includes should be after internal includes (`SDL.h` should come after).

File: src/platform/windows/backtrace_windows.c [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+#if defined(_WIN32) && defined(_M_X64)
+ 
+#include <windows.h>
+#include <imagehlp.h>
+#include <stdio.h>
+
+const char *print_exception_name(DWORD exceptionCode)
+{
+  switch (exceptionCode)
```
Comment (crudelios) [inline src/platform/windows/backtrace_windows.c:14]: 4 spaces identation please.

File: src/platform/windows/backtrace_windows.c [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+{
+  char crash_info[256];
+  int filelineinfo_ok = 0;
+  PIMAGEHLP_SYMBOL64 pSym;
+  STACKFRAME sf;
+  DWORD64 dwModBase, Disp64;
+  DWORD Disp;
+  BOOL more = FALSE;
+  IMAGEHLP_LINE64 line;
+  const char* filename = NULL;
```
Comment (crudelios) [inline src/platform/windows/backtrace_windows.c:58]:
```suggestion
  const char *filename = NULL;
```

**Review by crudelios — COMMENTED** [review]
Just a few nitpicks and small code formatting changes and I think it's ready to go.

**Comment by dalerank** [comment]
Fixed codestyle

File: .ci_scripts/install_dependencies.bat (hunk 1 above)
Comment (dalerank) [reply .ci_scripts/install_dependencies.bat]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (dalerank) [reply CMakeLists.txt]: Done

File: src/platform/platform.c (hunk 3 above)
Comment (dalerank) [reply src/platform/platform.c]: Done

File: .ci_scripts/install_dependencies.bat (hunk 4 above)
Comment (dalerank) [reply .ci_scripts/install_dependencies.bat]: Done

File: src/platform/windows/backtrace_windows.c (hunk 6 above)
Comment (dalerank) [reply src/platform/windows/backtrace_windows.c]: Done

File: src/platform/windows/backtrace_windows.c (hunk 8 above)
Comment (dalerank) [reply src/platform/windows/backtrace_windows.c]: Done

**Comment by crudelios** [comment]
I had a look at the code and it seems your changes allow us to remove the declaration of `backtrace_print`, which is only called from within `posix/backtrace.c`.

That also means the only needed declaration for `backtrace.h` is `install_game_crashhandler`, which makes calling the file `backtrace.c` a little bit misleading (it should become crash_handler.c` or something).

You see where this goes, it's a lot of minor code maintenance changes for no actual functional change.

As the code is good as-is, I can merge it and make the maintenance myself. Or you can do it yourself if you don't want me messing with your new code a little after you added it.

Just let me know what you prefer 🙂

**Comment by dalerank** [comment]
Sure, will be good. Because I cant test linux version now
Thanks

**Comment by crudelios** [comment]
All right, merged 👍

**Pull request merged**